Service crashes when adding a pin widget

```
java.lang.RuntimeException: Unable to stop activity {com.eblan.launcher/com.eblan.launcher.activity.main.MainActivity}: java.lang.IllegalArgumentException: Service not registered: com.eblan.launcher.feature.home.HomeScreenKt$LifecycleEffect$1$1$syncDataServiceConnection$1@4bcc331
	at android.app.ActivityThread.callActivityOnStop(ActivityThread.java:5365)
	at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:5337)
	at android.app.ActivityThread.handleStopActivity(ActivityThread.java:5402)
	at android.app.servertransaction.StopActivityItem.execute(StopActivityItem.java:43)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2335)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:241)
	at android.os.Looper.loop(Looper.java:342)
	at android.app.ActivityThread.main(ActivityThread.java:8118)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:583)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1045)
Caused by: java.lang.IllegalArgumentException: Service not registered: com.eblan.launcher.feature.home.HomeScreenKt$LifecycleEffect$1$1$syncDataServiceConnection$1@4bcc331
	at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1898)
	at android.app.ContextImpl.unbindService(ContextImpl.java:2167)
	at android.content.ContextWrapper.unbindService(ContextWrapper.java:863)
	at com.eblan.launcher.feature.home.HomeScreenKt.LifecycleEffect$lambda$0$0$0(HomeScreen.kt:1111)
	at com.eblan.launcher.feature.home.HomeScreenKt.$r8$lambda$IGrxcl30dsXpYZOhaSFlGWfZe1k(Unknown Source:0)
	at com.eblan.launcher.feature.home.HomeScreenKt$$ExternalSyntheticLambda19.onStateChanged(D8$$SyntheticClass:0)
	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.jvm.kt:316)
	at androidx.lifecycle.LifecycleRegistry.backwardPass(LifecycleRegistry.jvm.kt:268)
	at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.jvm.kt:287)
	at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.jvm.kt:136)
	at androidx.lifecycle.LifecycleRegistry.setCurrentState(LifecycleRegistry.jvm.kt:102)
	at androidx.navigation.internal.NavBackStackEntryImpl.updateState$navigation_common_release(NavBackStackEntryImpl.kt:107)
	at androidx.navigation.internal.NavBackStackEntryImpl.handleLifecycleEvent$navigation_common_release(NavBackStackEntryImpl.kt:92)
	at androidx.navigation.NavBackStackEntry.handleLifecycleEvent(NavBackStackEntry.android.kt:116)
	at androidx.navigation.internal.NavControllerImpl.lifecycleObserver$lambda$1(NavControllerImpl.kt:153)
	at androidx.navigation.internal.NavControllerImpl.$r8$lambda$NoxMp6GNHB-Ld29Z6h5JLrOHxiQ(Unknown Source:0)
	at androidx.navigation.internal.NavControllerImpl$$ExternalSyntheticLambda8.onStateChanged(D8$$SyntheticClass:0)
	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.jvm.kt:316)
	at androidx.lifecycle.LifecycleRegistry.backwardPass(LifecycleRegistry.jvm.kt:268)
	at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.jvm.kt:287)
	at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.jvm.kt:136)
	at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.jvm.kt:119)
	at androidx.lifecycle.ReportFragment$Companion.dispatch$lifecycle_runtime(ReportFragment.android.kt:181)
	at androidx.lifecycle.ReportFragment$LifecycleCallbacks.onActivityPreStopped(ReportFragment.android.kt:129)
	at android.app.Activity.dispatchActivityPreStopped(Activity.java:1457)
	at android.app.Activity.performStop(Activity.java:8328)
	at android.app.ActivityThread.callActivityOnStop(ActivityThread.java:5357)
	... 14 more

```

--- a/feature/home/src/main/kotlin/com/eblan/launcher/feature/home/HomeScreen.kt
+++ b/feature/home/src/main/kotlin/com/eblan/launcher/feature/home/HomeScreen.kt
@@ -1033,9 +1033,9 @@ private fun LifecycleEffect(
         val eblanNotificationListenerIntent =
             Intent(context, EblanNotificationListenerService::class.java)
 
-        var isSyncDataServiceBound = false
+        var shouldUnbindSyncDataService = false
 
-        var isEblanNotificationListenerServiceBound = false
+        var shouldUnbindEblanNotificationListenerService = false
 
         val syncDataServiceConnection = object : ServiceConnection {
             override fun onServiceConnected(name: ComponentName, service: IBinder) {
@@ -1048,18 +1048,15 @@ private fun LifecycleEffect(
                         }
                     }
                 }
-
-                isSyncDataServiceBound = true
             }
 
-            override fun onServiceDisconnected(name: ComponentName) {
-                isSyncDataServiceBound = false
-            }
+            override fun onServiceDisconnected(name: ComponentName) {}
         }
 
         val eblanNotificationListenerServiceConnection = object : ServiceConnection {
             override fun onServiceConnected(name: ComponentName, service: IBinder) {
-                val listener = (service as EblanNotificationListenerService.LocalBinder).getService()
+                val listener =
+                    (service as EblanNotificationListenerService.LocalBinder).getService()
 
                 scope.launch {
                     lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
@@ -1068,13 +1065,9 @@ private fun LifecycleEffect(
                         }
                     }
                 }
-
-                isEblanNotificationListenerServiceBound = true
             }
 
-            override fun onServiceDisconnected(name: ComponentName) {
-                isEblanNotificationListenerServiceBound = false
-            }
+            override fun onServiceDisconnected(name: ComponentName) {}
         }
 
         val lifecycleEventObserver = LifecycleEventObserver { _, event ->
@@ -1082,13 +1075,14 @@ private fun LifecycleEffect(
                 Lifecycle.Event.ON_START -> {
                     if (syncDataEnabled && pinItemRequestWrapper.getPinItemRequest() == null) {
                         context.startService(launcherAppsIntent)
-                        context.bindService(
+
+                        shouldUnbindSyncDataService = context.bindService(
                             syncDataIntent,
                             syncDataServiceConnection,
                             Context.BIND_AUTO_CREATE,
                         )
 
-                        context.bindService(
+                        shouldUnbindEblanNotificationListenerService = context.bindService(
                             eblanNotificationListenerIntent,
                             eblanNotificationListenerServiceConnection,
                             Context.BIND_AUTO_CREATE,
@@ -1100,11 +1094,11 @@ private fun LifecycleEffect(
 
                 Lifecycle.Event.ON_STOP -> {
                     if (syncDataEnabled && pinItemRequestWrapper.getPinItemRequest() == null) {
-                        if (isSyncDataServiceBound) {
+                        if (shouldUnbindSyncDataService) {
                             context.unbindService(syncDataServiceConnection)
                         }
 
-                        if (isEblanNotificationListenerServiceBound) {
+                        if (shouldUnbindEblanNotificationListenerService) {
                             context.unbindService(eblanNotificationListenerServiceConnection)
                         }
 
@@ -1121,10 +1115,6 @@ private fun LifecycleEffect(
         lifecycleOwner.lifecycle.addObserver(lifecycleEventObserver)
 
         onDispose {
-            isSyncDataServiceBound = false
-
-            isEblanNotificationListenerServiceBound = false
-
             lifecycleOwner.lifecycle.removeObserver(lifecycleEventObserver)
         }
     }

